react-frontend/src/ProfilePopup.js

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import Modal from "react-bootstrap/Modal";
+import "bootstrap/dist/css/bootstrap.min.css";
+
+function ProfilePopup(props) {
+  const [oldPassword, setOldPassword] = useState(null);
+  const [newPassword, setNewPassword] = useState(null);
+  const [confirmedNewPassword, setConfirmedNewPassword] = useState(null);
+
+  function handleChange(event) {
+    const { name, value } = event.target;
+    if (name === "oldPassword") setOldPassword(value);
+    else if (name === "newPassword") setNewPassword(value);
+    else setConfirmedNewPassword(value);
+  }
+
+  return (
+    <Modal
+      show={true}
+      onHide={() => {
+        props.setShow(false);
+      }}
+    >
+      <Modal.Header>
+        <button
+          className="close"
+          data-dismiss="modal"
+          aria-hidden="true"
+          onClick={() => {
+            props.setShow(false);
+          }}
+        >
+          ×
+        </button>
+      </Modal.Header>
+      <Modal.Body>
+        <div>
+          <form>
+            <label htmlFor="oldPassword">Old Password</label>
+            <input
+              type="password"
+              name="oldPassword"
+              id="oldPassword"
+              defaultValue={oldPassword}
+              onChange={handleChange}
+            />
+
+            <label htmlFor="newPassword">New Password</label>
+            <input
+              type="password"
+              name="newPassword"
+              id="newPassword"
+              defaultValue={newPassword}
+              onChange={handleChange}
+            />
+
+            <label htmlFor="confirmedNewPassword">Confirm New Password</label>
+            <input
+              type="password"
+              name="confirmedNewPassword"
+              id="confirmedNewPassword"
+              defaultValue={confirmedNewPassword}
+              onChange={handleChange}
+            />
+          </form>
+        </div>
+      </Modal.Body>
+      <Modal.Footer>
+        <input
+          type="button"
```

Comment:
instead of <input type="button"... 
you may want to use <button type="submit"...
This was used in the signup/login, and I think that that might be what is needed for just pressing enter to submit instead of clicking the button.

Reply:
Or nevermind, cause apparently that doesn't make that effect. no idea how Dustin did it then